Brocfile.js

```diff
@@ -1,202 +1,143 @@
 
-/*
- * configuration
- */
 var dirs = {
-  src: "",
-  js: "js",
-  jsDist: ".", // use . for root
-  styles: "css",
-  stylesDist: ".", // use . for root
-  img: "img",
+  src: "./src",
+  js: "./src/js",
+  dist: "",
+  styles: "./src/css",
+  img: "./src/img",
   imgDist: "img"
 };
 
-// without extensions
-var fileNames = {
+var files = {
   mainJs: "main",
   mainJsDist: "main",
-  mainStyles: "main",
-  mainStylesDist: "main"
+  mainLess: "main",
+  mainCssDist: "main",
+  index: "index.html"
 };
 
-/*
- * Task definitions
- */
 var tasks = {
-
-  jsHint: function (jsTree) {
-    // run jscs on compiled js
-    var jscsTree = jscs(jsTree, {
-      disableTestGenerator: true,
-      enabled: true,
-      logError: function (message) {
-        switch (env) {
-          case "production":
-            // TODO(mlunoe) disabled for production
-            break;
-          default:
-            // use pretty colors in test and development mode
-            console.log(chalk.red(message) + "\n");
-            break;
-        }
-      },
-      jshintrcPath: dirs.js + "/.jscsrc"
+  createJsTree: function () {
+    var jsTree = funnel(dirs.js, {
+      include: ["**/*.js", "**/*.jsx"],
+      destDir: dirs.js
     });
 
-    // run jshint on compiled js
-    var hintTree = jsHintTree(jsTree , {
-      logError: function (message) {
-        switch (env) {
-          case "production":
-            // TODO(mlunoe) disabled for production
-            break;
-          default:
-            // use pretty colors in test and development mode
-            this._errors.push(chalk.red(message) + "\n");
-            break;
-        }
-      },
-      jshintrcPath: dirs.js + "/.jshintrc"
-    });
+    return jsTree;
+  },
 
-    // hack to strip test files from jshint tree
-    hintTree = pickFiles(hintTree, {
-      srcDir: "/",
-      files: []
+  eslint: function (masterTree) {
+    return eslint(masterTree, {
+      config: ".eslintrc"
     });
-
-    return mergeTrees(
-      [jscsTree, hintTree, jsTree],
-      { overwrite: true }
-    );
   },
 
   webpack: function (masterTree) {
-    // transform merge module dependencies into one file
-    return webpackify(masterTree, {
-      entry: "./" + fileNames.mainJs + ".js",
+    var options = {
+      entry: dirs.js + "/" + files.mainJs + ".jsx",
       output: {
-        filename: dirs.jsDist + "/" + fileNames.mainJsDist + ".js"
+        filename: dirs.dist + "/" + files.mainJsDist + ".js"
+      },
+      module: {
+        loaders: [
+          {
+            test: /\.jsx$/,
+            loader: "jsx-loader?harmony",
+            exclude: /node_modules/
+          }
+        ],
+        postLoaders: [
+          {
+            loader: "transform?envify"
+          }
+        ]
+      },
+      resolve: {
+        extensions: ["", ".jsx", ".js"]
       }
-    });
+    };
+
+    // Extend options with source mapping
+    if (env === "development" && !process.env.DISABLE_SOURCE_MAP) {
```

Comment:
I think you answered my question about the index.html here, but what about the source map?